ClientManagementService/ClientManagementService.Infrastructure/Persistence/ClientRepository.cs

```diff
@@ -47,11 +56,36 @@ public async Task DeleteClientById(long id)
             }
         }
 
-        public async Task<List<Client>> GetAllClients()
+        public async Task<(List<Client>, int)> GetAllClientsByKeyword(int page = 1, int offset = 10, string keyword = "")
         {
             using (var context = new RofSchedulerContext())
             {
-                return await context.Clients.ToListAsync();
+                var skip = (page - 1) * offset;
+                IQueryable<Client> client = context.Clients;
+
+                if (!string.IsNullOrEmpty(keyword))
+                {
+                    keyword = keyword.ToLower();
+
+                    client = context.Clients
+                        .Where(e => (e.FirstName.ToLower().Contains(keyword))
+                            || (e.LastName.ToLower().Contains(keyword))
+                            || (e.EmailAddress.ToLower().Contains(keyword)));
+                }
+
+                var countByCriteria = await client.CountAsync();

```

Comment:
we should add if statement after this:

if (countByCriteria == 0){
  return empty list of clients.
}

right now, on initial load, if you delete all clients from DB, you get an exception.